domain/application/resource/types.go

```diff
@@ -110,13 +110,6 @@ type UnitResources struct {
 
 	// Resources are the resource versions currently in use by this unit.
 	Resources []Resource
-
-	// DownloadProgress indicates the number of bytes of the unit's
-	// resources, identified by name, that have been downloaded so far
-	// by the uniter. This only applies to resources that are currently
-	// being downloaded to the unit. All other resources for the unit
-	// will not be found in the map.
-	DownloadProgress map[string]int64
```

Comment:
Is this a strategic decision? 

While storing download progress in state is questionable, we do do it today; see `setUnitResourceProgress`.

Reply:
I raised this concern with @hmlanigan when I reviewed the first version of DDL. 

I wondered what where the use case behind that, because using a write clause on a db to report progress may be very prone to performance issue (we let to the caller an opportunity to spam our db with write request for any tiny progress he wants).

If i remember well, Heather states that we drop this functionality, since there is no such thing as "big resource" where this information can be useful.

I think it is a good decision. If we really need that one (for instance, to display to an user the download progress on each unit through a command line option ), I think we would better try to thing on a way to *compute* the progress instead of *report* it. As far I understand, progress is not a state, so it has nothing to do with the db (it is not persistent by essence)

However, if during wiring resource task we hit an issue impacted by this decision, we may need to reintroduce it.

Indeed, dropping this code now allows to avoid dead code later if this decision survive the full implementation of resource, and reintroduce it will be cheap (the real task will be on how to compute/store/whatever).